projects/igniteui-angular/src/lib/date-range/igx-date-range.component.ts

```diff
@@ -0,0 +1,506 @@
+})
+export class IgxDateRangeComponent implements IToggleView, AfterViewInit, OnDestroy, ControlValueAccessor {
+    /**
+     * Property which sets whether `IgxDateRangeComponent` is in dialog or dropdown mode.
+     *
+     * ```html
+     * <igx-date-range [mode]="'dropdown'"></igx-date-range
+     * ```
+     */
+    @Input()
+    public mode = InteractionMode.Dialog;
+
+    /**
+     * Property which sets the number displayed month views.
+     * Default is `2`.
+     *
+     * ```html
+     * <igx-date-range [monthsViewNumber]="3"></igx-date-range>
+     * ```
+     */
+    @Input()
+    public monthsViewNumber = 2;
+
+    /**
+     * Property which sets whether dates that are not part of the current month will be displayed.
+     * Default value is `false`.
+     */
+    @Input()
+    public hideOutsideDays: boolean;
+
+    /**
+     * Property which sets the start day of the week.
+     * Can be assigned to a numeric value or to `WEEKDAYS` enum value.
+     *
+     * ```html
+     * <igx-date-range [weekStart]="1"></igx-date-range>
+     * ```
+     */
+    @Input()
+    public weekStart = WEEKDAYS.SUNDAY;
+
+    /**
+     * Property which gets the `locale` of the calendar.
+     * Default value is `"en"`.
+     *
+     * ```html
+     * <igx-date-range [locale]="'jp'"></igx-date-range>
+     * ```
+     */
+    @Input()
+    public locale = 'en';
+
+    /**
+     * Property that applies a custom formatter function on the selected or passed in date.
+     *
+     * ```typescript
+     * private dayFormatter = new Intl.DateTimeFormat("en", { weekday: "long" });
+     * private monthFormatter = new Intl.DateTimeFormat("en", { month: "long" });
+     *
+     * public formatter(date: Date): string {
+     *  return `${this.dayFormatter.format(date)} - ${this.monthFormatter.format(date)} - ${date.getFullYear()}`;
+     * }
+     * ```
+     *
+     * ```html
+     * <igx-date-range [formatter]="formatter"></igx-date-range>
+     * ```
+     */
+    @Input()
+    public formatter: (val: DateRange) => string;
+
+    /**
+     * Property that changes the default text of the `done` button.
+     * It will show up only in `dialog` mode.
+     * Default value is `Done`.
+     *
+     * ```html
+     * <igx-date-range [doneButtonText]="'完了'"></igx-date-range>
+     * ```
+     */
+    @Input()
+    public doneButtonText = 'Done'; // optional
+
+    /**
+     * Property that changes the default overlay settings used by the `IgxDateRangeComponent`.
+     *
+     * ```html
+     * <igx-date-range [overlaySettings]="customOverlaySettings"></igx-date-range>
+     * ```
+     */
+    @Input()
+    public overlaySettings: OverlaySettings;
+
+    // TODO
+    @Input()
+    public displayFormat: string;
+
+    @Input()
+    public inputFormat: string;
+
+    /**
+     * An event that is emitted when a range is selected in the `IgxDateRangeComponent`.
+     */
+    @Output()
+    public onSelected = new EventEmitter<DateRange>();
```

Comment:
Has it been discussed if there should be a cancellable Selection event?
Leaving a comment on the PR, so others joining in can easily find it

Reply:
If this is to remain as is, the name should be changed to `rangeSelected` or smth